lib/credentials/apiKeyCredentials.ts

```diff
@@ -75,4 +76,41 @@ export class ApiKeyCredentials implements ServiceClientCredentials {
 
     return Promise.resolve(webResource);
   }
+
+  /**
+   * Signs a request with the values provided in the inHeader and inQuery parameter.
+   *
+   * @param httpRequest The HttpRequest to be signed.
+   * @returns The signed request.
+   */
+  public async signHttpRequest(httpRequest: HttpRequest): Promise<HttpRequest> {
+    if (!httpRequest) {
+      throw new Error(`httpRequest cannot be null or undefined and must be of type "object".`);
```

Comment:
Did this previously contain a `typeof` test? Also, I sorta wish that `strictNullChecks` would be adequate here, as TypeScript is supposed to warn if you ever pass a value that could be null/undefined to this method.

Reply:
I figure that in practice, crossing library boundaries and grabbing values over the wire makes this kind of thing more necessary..

Reply:
If we could guarantee that our customers would use TypeScript, then the type system would be enough here and I wouldn't need to check whether httpRequest is defined or not. For some of the request policies I tried to copy and paste the existing filter implementation with minimal changes. This is one of them.